cli/batch/sql.go

```diff
@@ -175,10 +175,26 @@ func buildBulkInsert(tbl *dax.Table, fields []*dax.Field, ids []interface{}, row
 	sb.WriteString(strings.Join(flds, ","))
 
 	// MAP
-	keyType := dax.BaseTypeID
-	if tbl.StringKeys() {
-		keyType = dax.BaseTypeString
+
+	// We need to set ID value below based on ids. Rather than changing what we
+	// get by changing IDK, decide how to format based on the type of ids. We
+	// get []uint64 for not keyed indexes or []byte for keyed indexes.
+	// Additionally, IDK doesn't give us a valid table for some reason (the keys
+	// options is always false). We will also use ids type to determine how to
+	// map values in the MAP clause
+	var fmtStr string
+	var keyType string
+	if len(ids) > 0 {
+		switch ids[0].(type) {
+		case uint64:
+			fmtStr = "%d"
+			keyType = dax.BaseTypeID
+		case []byte:
+			fmtStr = "%s"
+			keyType = dax.BaseTypeString
+		}
```

Comment:
A few things I wonder here.

- My first thought was "why is IDK giving us an invalid table and can we fix that instead of adding all of this switch logic?"
- If we do keep this switch, there should probably be a default case which returns an error for an unexpected type.
- The comment confused me at first because it mentions `[]uint64` and `[]byte`, but then handles `uint64` and `[]byte`. Is the comment meant to read `[][]byte`?

Reply:
- I'll take a look at the IDK stuff now. I spent a hour or so trying to figure it out but didn't get to the bottom of it. 
- Good call on the default case. I'll keep that in mind if I change this.
- Thanks for call this out. Yes, it should be `[][]byte`.

Reply:
I found a solution that is odd. Basically, IDK is incorrectly setting config options for the index it tracks. Then dax creates a table that can have the incorrect configuration based on this.

More involved answer, in idk/ingest.go, in the main.clone() function, we create an index with default configuration. Meaning keys is set to false. I don't know if this was a problem prior to the dax changes there but IDK doesn't seem to mind. Anyways, then, when we create a SQL batch, we generate the dax table. We do this by calling IndexInfoToTable(). This sets the type of `_id` based on whether or not the index IDK was tracking had keys or not (which is keys is always false). Thus, tbl.StringKeys() always return false in batching code of the kafka runner.

Reply:
I'd like to understand this a bit more. Can you tell where the table/index options are incorrectly set to begin with? One thing to understand is whether this test is being run against serverless or classic FeatureBase. That will determine which implementation of the `SchemaManager` is being used in the `idk.Main.clone()` method. It sounds to me like the call to `Schema()` there does not contain the table, and I'd like to understand why.

Reply:
Yes, that's correct. It's running against classic FeatureBase. I was assuming that meant I was supposed to have a NoOpSchemaManager because that's what I have. But maybe that's wrong? And yes, `index = schema.Index(m.Index)` returns an index with all defaults set. So, essentially, it's not going anywhere to get a schema of an index anywhere.

Reply:
Ah, normal ingeters setup their client which becomes the schema manager... https://github.com/FeatureBaseDB/featurebase/blob/master/idk/ingest.go#L859. Making that change and will see what happens.

Reply:
Yep, this fixed things & avro tests are passing now. thanks for the questions @travisturner!